plugins/auth/index.js

```diff
@@ -80,16 +82,30 @@ exports.register = (server, options, next) => {
     });
 
     /**
-     * Generates a jwt that is signed and has a 2h lifespan
+     * Generates a jwt that is signed and has a lifespan (default:2h, max:12h)
      * @method generateToken
-     * @param  {Object} profile Object from generateProfile
-     * @return {String}         Signed jwt that includes that profile
+     * @param  {Object}  profile     Object from generateProfile
+     * @param  {Integer} expiresSec  JWT Expires time (must be seconds)
+     * @return {String}              Signed jwt that includes that profile
      */
-    server.expose('generateToken', profile => jwt.sign(profile, pluginOptions.jwtPrivateKey, {
-        algorithm: ALGORITHM,
-        expiresIn: EXPIRES_IN,
-        jwtid: uuid()
-    }));
+    server.expose('generateToken', (profile, expiresSec = DEFAULT_EXPIRES_IN) => {
+        const executor = server.root.app.buildFactory.executor;
+
+        // max build timeout is set only when executor is 'k8s' or 'k8s-vm'
+        const maxExpiresIn = (executor.kubernetes && executor.kubernetes.maxBuildTimeout)
```

Comment:
can we base our logic on whether build timeout is present or not ? rather than explicitly based on executor value.
```
const buildTimeOut = getBuildTimeout(executor); 
const maxExpiresIn = buildTimeOut ? buildTimeOut * 60 : DEFAULT_MAX_EXPIRES_IN
```
`getBuildTimeout` can hide the details of explicit checks, without overloading this function.

Reply:
I fixed to use `getBuildTimeout()` method in models and use `executor` there, so it will work even with `executor-queue` .
And I also fixed the logic of `getBuildTimeout()` method.